internal/datastore/crdb/pool/balancer_test.go

```diff
@@ -0,0 +1,115 @@
+package pool
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodeConnectionBalancerPrune(t *testing.T) {
+	tests := []struct {
+		name       string
+		maxConns   uint32
+		nodes      []uint32
+		conns      []uint32
+		expectedGC []uint32
+	}{
+		{
+			name:       "no extra, no pruning needed",
+			nodes:      []uint32{1, 2, 3},
+			maxConns:   9,
+			conns:      []uint32{1, 1, 1, 2, 2, 2, 3, 3, 3},
+			expectedGC: []uint32{},
+		},
+		{
+			name:       "1 extra, prune 1",
+			nodes:      []uint32{1, 2, 3},
+			maxConns:   9,
+			conns:      []uint32{1, 1, 1, 1, 2, 3},
+			expectedGC: []uint32{1},
+		},
```

Comment:
can we add tests with maxConns == 1? And maybe potentially 2? To test the "halving" of connections to prune

Reply:
added a bunch more test cases with smaller max conns, but the halving is actually tested by the bigger tests with lots of connections